tests/test_exec.py

```diff
@@ -62,11 +64,71 @@ def test_exec_not_exists():
 
 def test_exec_detach_not_exists():
     return test_exec_not_exists_helper(False)
+
+NONKM_OK_CONF = "var/lib/krun/nonkm_exec_ok_config"
+
+def test_exec_non_km_payload_helper(execpath, nonkmok):
+    # only run these tests for krun
+    runtime = os.getenv("OCI_RUNTIME")
+    if os.path.basename(runtime) != "krun":
+        return 77  # skip this test if not using krun
+
+    conf = base_config()
+    conf['process']['args'] = ['/init', 'pause']
+    add_all_namespaces(conf)
+
+    # create allowed nonkm executables config file
+    nonkmok_file = "/tmp/nonkmok_config"
+    with open(nonkmok_file, "w") as c:
+        for nonkm_entry in nonkmok:
+            with open(nonkm_entry[2],"rb") as f:
+                bytes = f.read() # read entire file as bytes
+                readable_hash = hashlib.sha256(bytes).hexdigest()
+                c.write(nonkm_entry[0] + ":" + nonkm_entry[1] + ":" + readable_hash + "\n")
+
```

Comment:
btw - please ignore lines starting with '#' in you file parser, so we can add comments

Reply:
This is already being done.